trio/_core/_run.py

```diff
@@ -792,15 +792,19 @@ def cancel_called(self) -> bool:
         cancelled, then :attr:`cancelled_caught` is usually more
         appropriate.
         """
-        if self._cancel_status is not None or not self._has_been_entered:
+        if (
+            self._cancel_status is not None
+            or not self._has_been_entered
+            and not self._cancel_called
+            and current_time() >= self._deadline
+        ):
             # Scope is active or not yet entered: make sure cancel_called
             # is true if the deadline has passed. This shouldn't
             # be able to actually change behavior, since we check for
             # deadline expiry on scope entry and at every checkpoint,
             # but it makes the value returned by cancel_called more
             # closely match expectations.
```

Comment:
Comment applies to the 2 conditions in the `if` statement, not the code

Reply:
Reverted as of 76d741357e82eed4ffd8de3e540804ca5346a5b3

Reply:
Reverted as of most recent commit